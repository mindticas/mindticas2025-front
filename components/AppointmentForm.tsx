'use client';
import {
    Button,
    Fieldset,
    Flex,
    Input,
    Spinner,
    Stack,
} from '@chakra-ui/react';
import { Field } from '@/components/ui/field';
import React, { useState } from 'react';
import PhoneInput, { Value } from 'react-phone-number-input';
import { isValidPhoneNumber } from 'react-phone-number-input';
import 'react-phone-number-input/style.css';
import { useBookingContext } from '@/context/BookingContext';
import { toaster } from './ui/toaster';
import { createAppointment } from '@/services/AppointmentService';

interface AppointmentFormProps {
    onSuccess: () => void;
    onError: () => void;
}

export default function AppointmentForm({
    onSuccess,
    onError,
}: AppointmentFormProps) {
    const {
        treatment,
        personData,
        setPersonData,
        dateTime,
        setIsSuccessfullyBooked,
    } = useBookingContext();

    // State for loading state
    const [isLoading, setIsLoading] = useState(false);
    // State for phone number validation
    const [isPhoneValid, setIsPhoneValid] = useState(true);

    // Handle phone number change
    const handlePhoneChange = (phone: Value) => {
        setPersonData({
            ...personData,
            phone: phone ? phone.toString() : '',
        });

        // Validate phone number
        if (phone) {
            setIsPhoneValid(isValidPhoneNumber(phone));
        } else {
            setIsPhoneValid(true); // Reset validation if empty
        }
    };

    // Handle input changes and update state
    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        const { name, value } = e.target;
        setPersonData({ ...personData, [name]: value });
    };

    // Function to remove country code (+52) and keep only 10 digits
    const formatPhoneNumber = (phone: string) => {
        if (phone.startsWith('+52')) {
            return phone.slice(3); // Remove +52
        }
        return phone; // If no country code, return as is
    };

    // Handle form submission
    const handleSubmit = async (e: React.FormEvent) => {
        e.preventDefault();

        // Validate phone number before submission
        if (!personData.phone || !isValidPhoneNumber(personData.phone)) {
            toaster.create({
                type: 'error',
                duration: 5000,
                title: 'Número de teléfono inválido o vacío',
            });
            return;
        }

        // Validate form fields before submission
        if (!personData.name || !personData.lastName) {
            onError();
            return;
        }

        // Set loading state spinner
        setIsLoading(true);

        try {
            // Format phone number to remove country code
            const formattedPhone = formatPhoneNumber(personData.phone);
            // Create object with appointment data
            const newAppointment = {
                name: `${personData.name} ${personData.lastName}`,
                phone: formattedPhone,
                treatment_ids: [treatment],
                scheduledStart: dateTime,
            };

            // create appointment in the backend
            await createAppointment(newAppointment);

            // if appointment is created successfully, show success message
            setIsLoading(false);
            onSuccess();
            setIsSuccessfullyBooked(true);
            setPersonData({ name: '', lastName: '', phone: '' });
        } catch (error) {
            setIsLoading(false);
            onError();
        }
    };

    return (
        <form onSubmit={handleSubmit} style={{ width: '100%' }}>
            <Flex p='6' justify='center'>
                <Fieldset.Root
                    borderRadius='lg'
                    boxShadow='md'
                    borderWidth='1px'
                    borderColor='gray.200'
                    size='lg'
                    maxW='sm'
                    p={6}
                >
                    <Stack>
                        <Fieldset.Legend
                            fontSize='lg'
                            fontWeight='600'
                            color='gray.900'
                        >
                            Datos del contacto
                        </Fieldset.Legend>
                        <Fieldset.HelperText color='gray.600' pb={3}>
                            Ingrese los datos del contacto para agendar una cita
                        </Fieldset.HelperText>
                    </Stack>
                    <Fieldset.Content>
                        <Field required label='Nombre'>
                            <Input
                                size='lg'
                                p={1}
                                type='text'
                                name='name'
                                placeholder='Juan'
                                value={personData.name}
                                onChange={handleChange}
                            />
                        </Field>
                        <Field required label='Apellido'>
                            <Input
                                size='lg'
                                p={1}
                                name='lastName'
                                type='text'
                                placeholder='Torres'
                                value={personData.lastName}
                                onChange={handleChange}
                            />
                        </Field>
                        <Field required label='Número de teléfono'>
                            <PhoneInput
                                className='PhoneInputInput'
                                countries={['MX']}
                                defaultCountry='MX'
                                placeholder='Ingresa tu número de teléfono'
                                value={personData.phone}
                                onChange={handlePhoneChange}
                                style={{
                                    padding: '8px',
                                    fontSize: '16px',
                                    border: isPhoneValid
                                        ? '1px solid #ccc'
                                        : '1px solid red',
                                    borderRadius: '4px',
                                    width: '100%',
                                }}
                            />
                            {!isPhoneValid && (
                                <p style={{ color: 'red', fontSize: '14px' }}>
                                    Número de teléfono inválido.
                                </p>
                            )}
                        </Field>
                    </Fieldset.Content>
                    <Button
                        bg='black'
                        color='white'
                        type='submit'
                        w='full'
                        mt={6}
                        disabled={isLoading}
                    >
                        {isLoading ? <Spinner size='sm' mr={2} /> : 'Agendar'}
                    </Button>
                </Fieldset.Root>
            </Flex>
        </form>
    );
}
