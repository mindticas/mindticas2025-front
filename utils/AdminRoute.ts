import Layout from '@/components/Layout';
import { Box } from '@chakra-ui/react';

export class AdminRoute {
    // Fixed route for admin
    private static route: string = '/admin';

    // Returns the Layout component if the pathname contains '/admin', otherwise returns box
    static getLayout(pathname: string) {
        return pathname.includes(this.route) ? Layout : Box;
    }
}
