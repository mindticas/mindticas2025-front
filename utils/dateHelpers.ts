// Gets the name of the day according to the full calendar format
export const getDayName = (dateStr: string): string => {
    const [year, month, day] = dateStr.split('-').map(Number)
    const date = new Date(year, month -1, day)
    return date.toLocaleDateString('es-ES', { weekday: 'long'})
}
// Get the milliseconds in a day
export function addDays(date: Date, days: number): Date {
    const milisecondsPerDay = 24 * 60 * 60 * 1000; 
    return new Date(date.getTime() + days * milisecondsPerDay)
}