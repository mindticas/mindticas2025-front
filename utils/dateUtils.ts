import { DateTime } from "luxon";

interface DateTimeParts {
    date: string;
    time: string;
}

// * @returns Object with (YYYY-MM-DD) y time (HH:MM)

export const splitDateTimeFromISO = (
    isoString: string | null | undefined,
): DateTimeParts => {
    if (!isoString) return { date: '', time: '' };

    const dateObj = new Date(isoString);

    if (isNaN(dateObj.getTime())) {
        return { date: '', time: '' };
    }

    // format date (YYYY-MM-DD)
    const date = DateTime.fromJSDate(dateObj)
       .setZone('local')
        .toFormat('yyyy-MM-dd');

    // format time (HH:MM)
    const time = dateObj
        .toLocaleTimeString('es-ES', {
            hour: '2-digit',
            minute: '2-digit',
            hour12: false,
        })
        .replace(/\./g, '');

    return { date, time };
};
