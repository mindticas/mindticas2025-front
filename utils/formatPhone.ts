export default function formatPhone(phone: string): string {
    // Remove everything that is not digits
    const digits = phone.replace(/\D/g, '');
    // Applies format only if it has 10 digits
    if (digits.length === 10) {
        return `${digits.slice(0, 3)}-${digits.slice(3, 6)}-${digits.slice(6)}`;
    }
    return phone; // Returns the original if it doesn't match
}
