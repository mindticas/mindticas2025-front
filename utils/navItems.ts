import { BarChart, Calendar, Scissors, UserCog, Users } from 'lucide-react';

export const navItems = [
    { name: 'Citas', href: '/admin/', icon: Calendar },
    { name: 'Clientes', href: '/admin/clients', icon: Users },
    { name: 'Empleados', href: '/admin/employees', icon: UserCog },
    { name: 'Tratamientos', href: '/admin/treatments', icon: Scissors },
    { name: 'Reportes', href: '/admin/reports', icon: BarChart },
];
