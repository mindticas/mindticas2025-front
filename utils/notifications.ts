import { toaster } from '@/components/ui/toaster';

type ToastType = 'success' | 'error' | 'info' | 'warning';

interface ToastOption {
    duration?: number;
    title: string;
    description?: string;
}

const showToast = (type: ToastType, options: ToastOption) => {
    toaster.create({
        type,
        duration: options.duration || 5000,
        title: options.title,
        description: options.description,
    });
};

const TreatmentNotification = {
    createSuccess: () =>
        showToast('success', { title: 'Tratamiento creado con éxito' }),
    updateSuccess: () =>
        showToast('success', { title: 'Tratamiento actualizado con éxito' }),
    deleteSuccess: () =>
        showToast('success', { title: 'Tratamiento eliminado con éxito' }),
    createError: (message?: string) =>
        showToast('error', {
            title: message || 'Error al crear el tratamiento',
        }),
    updateError: (message?: string) =>
        showToast('error', {
            title: message || 'Error al actualizar el tratamiento0',
        }),
    deleteError: () =>
        showToast('error', { title: 'Error al eliminar el tratamiento' }),
    noChanges: () =>
        showToast('info', {
            title: 'No se detectaron cambios para actualizar',
        }),
};

export default TreatmentNotification;
