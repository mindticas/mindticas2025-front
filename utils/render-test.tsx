import { Provider } from '@/components/ui/provider';
import { BookingProvider } from '@/context/BookingContext';
import { render as rtlRender } from '@testing-library/react';

export function render(ui: React.ReactNode) {
    return rtlRender(<>{ui}</>, {
        wrapper: (props: React.PropsWithChildren) => (
            <Provider>
                <BookingProvider>{props.children}</BookingProvider>
            </Provider>
        ),
    });
}
