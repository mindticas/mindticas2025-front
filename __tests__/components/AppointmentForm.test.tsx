import { render } from '../../utils/render-test';
import AppointmentForm from '@/components/AppointmentForm';

describe('AppointmentForm Snapshot', () => {
    it('should match the snapshot', () => {
        const mockOnSuccess = jest.fn();
        const mockOnError = jest.fn();
        const { asFragment } = render(
            <AppointmentForm onSuccess={mockOnSuccess} onError={mockOnError} />,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
