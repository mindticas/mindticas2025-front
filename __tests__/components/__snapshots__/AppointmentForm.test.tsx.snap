// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Snapshot should match the snapshot 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <form
    style="width: 100%;"
  >
    <div
      class="css-s6kjdg"
    >
      <fieldset
        class="fieldset__root css-1pplgli"
        data-part="root"
        data-scope="fieldset"
      >
        <div
          class="chakra-stack css-8g8ihq"
        >
          <legend
            class="fieldset__legend css-181jjb6"
            data-part="legend"
            data-scope="fieldset"
          >
            Datos del contacto
          </legend>
          <span
            class="fieldset__helperText css-a5zrz7"
            data-part="helper-text"
            data-scope="fieldset"
            id="fieldset:::r0:::helper-text"
          >
            Ingrese los datos del contacto para agendar una cita
          </span>
        </div>
        <div
          class="fieldset__content css-mpjbgj"
        >
          <div
            class="chakra-field__root css-13f87cb"
            data-part="root"
            data-scope="field"
            id="field:::r1:"
            role="group"
          >
            <label
              class="chakra-field__label css-65zcwe"
              data-part="label"
              data-scope="field"
              for=":r1:"
              id="field:::r1:::label"
            >
              Nombre
              <span
                aria-hidden="true"
                class="chakra-field__requiredIndicator css-gp8wgo"
              >
                *
              </span>
            </label>
            <input
              class="chakra-input css-6w9nzz"
              data-part="input"
              data-required=""
              data-scope="field"
              id=":r1:"
              name="name"
              placeholder="Juan"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="chakra-field__root css-13f87cb"
            data-part="root"
            data-scope="field"
            id="field:::r2:"
            role="group"
          >
            <label
              class="chakra-field__label css-65zcwe"
              data-part="label"
              data-scope="field"
              for=":r2:"
              id="field:::r2:::label"
            >
              Apellido
              <span
                aria-hidden="true"
                class="chakra-field__requiredIndicator css-gp8wgo"
              >
                *
              </span>
            </label>
            <input
              class="chakra-input css-6w9nzz"
              data-part="input"
              data-required=""
              data-scope="field"
              id=":r2:"
              name="lastname"
              placeholder="Torres"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="chakra-field__root css-13f87cb"
            data-part="root"
            data-scope="field"
            id="field:::r3:"
            role="group"
          >
            <label
              class="chakra-field__label css-65zcwe"
              data-part="label"
              data-scope="field"
              for=":r3:"
              id="field:::r3:::label"
            >
              Número de teléfono
              <span
                aria-hidden="true"
                class="chakra-field__requiredIndicator css-gp8wgo"
              >
                *
              </span>
            </label>
            <div
              class="PhoneInputInput PhoneInput"
              style="padding: 8px; font-size: 16px; border: 1px solid #ccc; border-radius: 4px; width: 100%;"
            >
              <div
                class="PhoneInputCountry"
              >
                <select
                  aria-label="Phone number country"
                  class="PhoneInputCountrySelect"
                >
                  <option
                    value="ZZ"
                  >
                    International
                  </option>
                  <option
                    value="AF"
                  >
                    Afghanistan
                  </option>
                  <option
                    value="AX"
                  >
                    Åland Islands
                  </option>
                  <option
                    value="AL"
                  >
                    Albania
                  </option>
                  <option
                    value="DZ"
                  >
                    Algeria
                  </option>
                  <option
                    value="AS"
                  >
                    American Samoa
                  </option>
                  <option
                    value="AD"
                  >
                    Andorra
                  </option>
                  <option
                    value="AO"
                  >
                    Angola
                  </option>
                  <option
                    value="AI"
                  >
                    Anguilla
                  </option>
                  <option
                    value="AG"
                  >
                    Antigua and Barbuda
                  </option>
                  <option
                    value="AR"
                  >
                    Argentina
                  </option>
                  <option
                    value="AM"
                  >
                    Armenia
                  </option>
                  <option
                    value="AW"
                  >
                    Aruba
                  </option>
                  <option
                    value="AC"
                  >
                    Ascension Island
                  </option>
                  <option
                    value="AU"
                  >
                    Australia
                  </option>
                  <option
                    value="AT"
                  >
                    Austria
                  </option>
                  <option
                    value="AZ"
                  >
                    Azerbaijan
                  </option>
                  <option
                    value="BS"
                  >
                    Bahamas
                  </option>
                  <option
                    value="BH"
                  >
                    Bahrain
                  </option>
                  <option
                    value="BD"
                  >
                    Bangladesh
                  </option>
                  <option
                    value="BB"
                  >
                    Barbados
                  </option>
                  <option
                    value="BY"
                  >
                    Belarus
                  </option>
                  <option
                    value="BE"
                  >
                    Belgium
                  </option>
                  <option
                    value="BZ"
                  >
                    Belize
                  </option>
                  <option
                    value="BJ"
                  >
                    Benin
                  </option>
                  <option
                    value="BM"
                  >
                    Bermuda
                  </option>
                  <option
                    value="BT"
                  >
                    Bhutan
                  </option>
                  <option
                    value="BO"
                  >
                    Bolivia
                  </option>
                  <option
                    value="BQ"
                  >
                    Bonaire, Sint Eustatius and Saba
                  </option>
                  <option
                    value="BA"
                  >
                    Bosnia and Herzegovina
                  </option>
                  <option
                    value="BW"
                  >
                    Botswana
                  </option>
                  <option
                    value="BR"
                  >
                    Brazil
                  </option>
                  <option
                    value="IO"
                  >
                    British Indian Ocean Territory
                  </option>
                  <option
                    value="BN"
                  >
                    Brunei Darussalam
                  </option>
                  <option
                    value="BG"
                  >
                    Bulgaria
                  </option>
                  <option
                    value="BF"
                  >
                    Burkina Faso
                  </option>
                  <option
                    value="BI"
                  >
                    Burundi
                  </option>
                  <option
                    value="KH"
                  >
                    Cambodia
                  </option>
                  <option
                    value="CM"
                  >
                    Cameroon
                  </option>
                  <option
                    value="CA"
                  >
                    Canada
                  </option>
                  <option
                    value="CV"
                  >
                    Cape Verde
                  </option>
                  <option
                    value="KY"
                  >
                    Cayman Islands
                  </option>
                  <option
                    value="CF"
                  >
                    Central African Republic
                  </option>
                  <option
                    value="TD"
                  >
                    Chad
                  </option>
                  <option
                    value="CL"
                  >
                    Chile
                  </option>
                  <option
                    value="CN"
                  >
                    China
                  </option>
                  <option
                    value="CX"
                  >
                    Christmas Island
                  </option>
                  <option
                    value="CC"
                  >
                    Cocos (Keeling) Islands
                  </option>
                  <option
                    value="CO"
                  >
                    Colombia
                  </option>
                  <option
                    value="KM"
                  >
                    Comoros
                  </option>
                  <option
                    value="CG"
                  >
                    Congo
                  </option>
                  <option
                    value="CD"
                  >
                    Congo, Democratic Republic of the
                  </option>
                  <option
                    value="CK"
                  >
                    Cook Islands
                  </option>
                  <option
                    value="CR"
                  >
                    Costa Rica
                  </option>
                  <option
                    value="CI"
                  >
                    Cote d'Ivoire
                  </option>
                  <option
                    value="HR"
                  >
                    Croatia
                  </option>
                  <option
                    value="CU"
                  >
                    Cuba
                  </option>
                  <option
                    value="CW"
                  >
                    Curaçao
                  </option>
                  <option
                    value="CY"
                  >
                    Cyprus
                  </option>
                  <option
                    value="CZ"
                  >
                    Czech Republic
                  </option>
                  <option
                    value="DK"
                  >
                    Denmark
                  </option>
                  <option
                    value="DJ"
                  >
                    Djibouti
                  </option>
                  <option
                    value="DM"
                  >
                    Dominica
                  </option>
                  <option
                    value="DO"
                  >
                    Dominican Republic
                  </option>
                  <option
                    value="EC"
                  >
                    Ecuador
                  </option>
                  <option
                    value="EG"
                  >
                    Egypt
                  </option>
                  <option
                    value="SV"
                  >
                    El Salvador
                  </option>
                  <option
                    value="GQ"
                  >
                    Equatorial Guinea
                  </option>
                  <option
                    value="ER"
                  >
                    Eritrea
                  </option>
                  <option
                    value="EE"
                  >
                    Estonia
                  </option>
                  <option
                    value="ET"
                  >
                    Ethiopia
                  </option>
                  <option
                    value="FK"
                  >
                    Falkland Islands
                  </option>
                  <option
                    value="FO"
                  >
                    Faroe Islands
                  </option>
                  <option
                    value="FM"
                  >
                    Federated States of Micronesia
                  </option>
                  <option
                    value="FJ"
                  >
                    Fiji
                  </option>
                  <option
                    value="FI"
                  >
                    Finland
                  </option>
                  <option
                    value="FR"
                  >
                    France
                  </option>
                  <option
                    value="GF"
                  >
                    French Guiana
                  </option>
                  <option
                    value="PF"
                  >
                    French Polynesia
                  </option>
                  <option
                    value="GA"
                  >
                    Gabon
                  </option>
                  <option
                    value="GM"
                  >
                    Gambia
                  </option>
                  <option
                    value="GE"
                  >
                    Georgia
                  </option>
                  <option
                    value="DE"
                  >
                    Germany
                  </option>
                  <option
                    value="GH"
                  >
                    Ghana
                  </option>
                  <option
                    value="GI"
                  >
                    Gibraltar
                  </option>
                  <option
                    value="GR"
                  >
                    Greece
                  </option>
                  <option
                    value="GL"
                  >
                    Greenland
                  </option>
                  <option
                    value="GD"
                  >
                    Grenada
                  </option>
                  <option
                    value="GP"
                  >
                    Guadeloupe
                  </option>
                  <option
                    value="GU"
                  >
                    Guam
                  </option>
                  <option
                    value="GT"
                  >
                    Guatemala
                  </option>
                  <option
                    value="GG"
                  >
                    Guernsey
                  </option>
                  <option
                    value="GN"
                  >
                    Guinea
                  </option>
                  <option
                    value="GW"
                  >
                    Guinea-Bissau
                  </option>
                  <option
                    value="GY"
                  >
                    Guyana
                  </option>
                  <option
                    value="HT"
                  >
                    Haiti
                  </option>
                  <option
                    value="VA"
                  >
                    Holy See (Vatican City State)
                  </option>
                  <option
                    value="HN"
                  >
                    Honduras
                  </option>
                  <option
                    value="HK"
                  >
                    Hong Kong
                  </option>
                  <option
                    value="HU"
                  >
                    Hungary
                  </option>
                  <option
                    value="IS"
                  >
                    Iceland
                  </option>
                  <option
                    value="IN"
                  >
                    India
                  </option>
                  <option
                    value="ID"
                  >
                    Indonesia
                  </option>
                  <option
                    value="IR"
                  >
                    Iran
                  </option>
                  <option
                    value="IQ"
                  >
                    Iraq
                  </option>
                  <option
                    value="IE"
                  >
                    Ireland
                  </option>
                  <option
                    value="IM"
                  >
                    Isle of Man
                  </option>
                  <option
                    value="IL"
                  >
                    Israel
                  </option>
                  <option
                    value="IT"
                  >
                    Italy
                  </option>
                  <option
                    value="JM"
                  >
                    Jamaica
                  </option>
                  <option
                    value="JP"
                  >
                    Japan
                  </option>
                  <option
                    value="JE"
                  >
                    Jersey
                  </option>
                  <option
                    value="JO"
                  >
                    Jordan
                  </option>
                  <option
                    value="KZ"
                  >
                    Kazakhstan
                  </option>
                  <option
                    value="KE"
                  >
                    Kenya
                  </option>
                  <option
                    value="KI"
                  >
                    Kiribati
                  </option>
                  <option
                    value="XK"
                  >
                    Kosovo
                  </option>
                  <option
                    value="KW"
                  >
                    Kuwait
                  </option>
                  <option
                    value="KG"
                  >
                    Kyrgyzstan
                  </option>
                  <option
                    value="LA"
                  >
                    Laos
                  </option>
                  <option
                    value="LV"
                  >
                    Latvia
                  </option>
                  <option
                    value="LB"
                  >
                    Lebanon
                  </option>
                  <option
                    value="LS"
                  >
                    Lesotho
                  </option>
                  <option
                    value="LR"
                  >
                    Liberia
                  </option>
                  <option
                    value="LY"
                  >
                    Libya
                  </option>
                  <option
                    value="LI"
                  >
                    Liechtenstein
                  </option>
                  <option
                    value="LT"
                  >
                    Lithuania
                  </option>
                  <option
                    value="LU"
                  >
                    Luxembourg
                  </option>
                  <option
                    value="MO"
                  >
                    Macao
                  </option>
                  <option
                    value="MG"
                  >
                    Madagascar
                  </option>
                  <option
                    value="MW"
                  >
                    Malawi
                  </option>
                  <option
                    value="MY"
                  >
                    Malaysia
                  </option>
                  <option
                    value="MV"
                  >
                    Maldives
                  </option>
                  <option
                    value="ML"
                  >
                    Mali
                  </option>
                  <option
                    value="MT"
                  >
                    Malta
                  </option>
                  <option
                    value="MH"
                  >
                    Marshall Islands
                  </option>
                  <option
                    value="MQ"
                  >
                    Martinique
                  </option>
                  <option
                    value="MR"
                  >
                    Mauritania
                  </option>
                  <option
                    value="MU"
                  >
                    Mauritius
                  </option>
                  <option
                    value="YT"
                  >
                    Mayotte
                  </option>
                  <option
                    value="MX"
                  >
                    Mexico
                  </option>
                  <option
                    value="MD"
                  >
                    Moldova
                  </option>
                  <option
                    value="MC"
                  >
                    Monaco
                  </option>
                  <option
                    value="MN"
                  >
                    Mongolia
                  </option>
                  <option
                    value="ME"
                  >
                    Montenegro
                  </option>
                  <option
                    value="MS"
                  >
                    Montserrat
                  </option>
                  <option
                    value="MA"
                  >
                    Morocco
                  </option>
                  <option
                    value="MZ"
                  >
                    Mozambique
                  </option>
                  <option
                    value="MM"
                  >
                    Myanmar
                  </option>
                  <option
                    value="NA"
                  >
                    Namibia
                  </option>
                  <option
                    value="NR"
                  >
                    Nauru
                  </option>
                  <option
                    value="NP"
                  >
                    Nepal
                  </option>
                  <option
                    value="NL"
                  >
                    Netherlands
                  </option>
                  <option
                    value="NC"
                  >
                    New Caledonia
                  </option>
                  <option
                    value="NZ"
                  >
                    New Zealand
                  </option>
                  <option
                    value="NI"
                  >
                    Nicaragua
                  </option>
                  <option
                    value="NE"
                  >
                    Niger
                  </option>
                  <option
                    value="NG"
                  >
                    Nigeria
                  </option>
                  <option
                    value="NU"
                  >
                    Niue
                  </option>
                  <option
                    value="NF"
                  >
                    Norfolk Island
                  </option>
                  <option
                    value="KP"
                  >
                    North Korea
                  </option>
                  <option
                    value="MK"
                  >
                    North Macedonia
                  </option>
                  <option
                    value="MP"
                  >
                    Northern Mariana Islands
                  </option>
                  <option
                    value="NO"
                  >
                    Norway
                  </option>
                  <option
                    value="OM"
                  >
                    Oman
                  </option>
                  <option
                    value="PK"
                  >
                    Pakistan
                  </option>
                  <option
                    value="PW"
                  >
                    Palau
                  </option>
                  <option
                    value="PS"
                  >
                    Palestine
                  </option>
                  <option
                    value="PA"
                  >
                    Panama
                  </option>
                  <option
                    value="PG"
                  >
                    Papua New Guinea
                  </option>
                  <option
                    value="PY"
                  >
                    Paraguay
                  </option>
                  <option
                    value="PE"
                  >
                    Peru
                  </option>
                  <option
                    value="PH"
                  >
                    Philippines
                  </option>
                  <option
                    value="PL"
                  >
                    Poland
                  </option>
                  <option
                    value="PT"
                  >
                    Portugal
                  </option>
                  <option
                    value="PR"
                  >
                    Puerto Rico
                  </option>
                  <option
                    value="QA"
                  >
                    Qatar
                  </option>
                  <option
                    value="RE"
                  >
                    Reunion
                  </option>
                  <option
                    value="RO"
                  >
                    Romania
                  </option>
                  <option
                    value="RU"
                  >
                    Russia
                  </option>
                  <option
                    value="RW"
                  >
                    Rwanda
                  </option>
                  <option
                    value="BL"
                  >
                    Saint Barthélemy
                  </option>
                  <option
                    value="SH"
                  >
                    Saint Helena
                  </option>
                  <option
                    value="KN"
                  >
                    Saint Kitts and Nevis
                  </option>
                  <option
                    value="LC"
                  >
                    Saint Lucia
                  </option>
                  <option
                    value="MF"
                  >
                    Saint Martin (French Part)
                  </option>
                  <option
                    value="PM"
                  >
                    Saint Pierre and Miquelon
                  </option>
                  <option
                    value="VC"
                  >
                    Saint Vincent and the Grenadines
                  </option>
                  <option
                    value="WS"
                  >
                    Samoa
                  </option>
                  <option
                    value="SM"
                  >
                    San Marino
                  </option>
                  <option
                    value="ST"
                  >
                    Sao Tome and Principe
                  </option>
                  <option
                    value="SA"
                  >
                    Saudi Arabia
                  </option>
                  <option
                    value="SN"
                  >
                    Senegal
                  </option>
                  <option
                    value="RS"
                  >
                    Serbia
                  </option>
                  <option
                    value="SC"
                  >
                    Seychelles
                  </option>
                  <option
                    value="SL"
                  >
                    Sierra Leone
                  </option>
                  <option
                    value="SG"
                  >
                    Singapore
                  </option>
                  <option
                    value="SX"
                  >
                    Sint Maarten
                  </option>
                  <option
                    value="SK"
                  >
                    Slovakia
                  </option>
                  <option
                    value="SI"
                  >
                    Slovenia
                  </option>
                  <option
                    value="SB"
                  >
                    Solomon Islands
                  </option>
                  <option
                    value="SO"
                  >
                    Somalia
                  </option>
                  <option
                    value="ZA"
                  >
                    South Africa
                  </option>
                  <option
                    value="KR"
                  >
                    South Korea
                  </option>
                  <option
                    value="SS"
                  >
                    South Sudan
                  </option>
                  <option
                    value="ES"
                  >
                    Spain
                  </option>
                  <option
                    value="LK"
                  >
                    Sri Lanka
                  </option>
                  <option
                    value="SD"
                  >
                    Sudan
                  </option>
                  <option
                    value="SR"
                  >
                    Suriname
                  </option>
                  <option
                    value="SJ"
                  >
                    Svalbard and Jan Mayen
                  </option>
                  <option
                    value="SZ"
                  >
                    Swaziland
                  </option>
                  <option
                    value="SE"
                  >
                    Sweden
                  </option>
                  <option
                    value="CH"
                  >
                    Switzerland
                  </option>
                  <option
                    value="SY"
                  >
                    Syria
                  </option>
                  <option
                    value="TW"
                  >
                    Taiwan
                  </option>
                  <option
                    value="TJ"
                  >
                    Tajikistan
                  </option>
                  <option
                    value="TZ"
                  >
                    Tanzania
                  </option>
                  <option
                    value="TH"
                  >
                    Thailand
                  </option>
                  <option
                    value="TL"
                  >
                    Timor-Leste
                  </option>
                  <option
                    value="TG"
                  >
                    Togo
                  </option>
                  <option
                    value="TK"
                  >
                    Tokelau
                  </option>
                  <option
                    value="TO"
                  >
                    Tonga
                  </option>
                  <option
                    value="TT"
                  >
                    Trinidad and Tobago
                  </option>
                  <option
                    value="TA"
                  >
                    Tristan da Cunha
                  </option>
                  <option
                    value="TN"
                  >
                    Tunisia
                  </option>
                  <option
                    value="TR"
                  >
                    Turkey
                  </option>
                  <option
                    value="TM"
                  >
                    Turkmenistan
                  </option>
                  <option
                    value="TC"
                  >
                    Turks and Caicos Islands
                  </option>
                  <option
                    value="TV"
                  >
                    Tuvalu
                  </option>
                  <option
                    value="UG"
                  >
                    Uganda
                  </option>
                  <option
                    value="UA"
                  >
                    Ukraine
                  </option>
                  <option
                    value="AE"
                  >
                    United Arab Emirates
                  </option>
                  <option
                    value="GB"
                  >
                    United Kingdom
                  </option>
                  <option
                    value="US"
                  >
                    United States
                  </option>
                  <option
                    value="UY"
                  >
                    Uruguay
                  </option>
                  <option
                    value="UZ"
                  >
                    Uzbekistan
                  </option>
                  <option
                    value="VU"
                  >
                    Vanuatu
                  </option>
                  <option
                    value="VE"
                  >
                    Venezuela
                  </option>
                  <option
                    value="VN"
                  >
                    Vietnam
                  </option>
                  <option
                    value="VG"
                  >
                    Virgin Islands, British
                  </option>
                  <option
                    value="VI"
                  >
                    Virgin Islands, U.S.
                  </option>
                  <option
                    value="WF"
                  >
                    Wallis and Futuna
                  </option>
                  <option
                    value="EH"
                  >
                    Western Sahara
                  </option>
                  <option
                    value="YE"
                  >
                    Yemen
                  </option>
                  <option
                    value="ZM"
                  >
                    Zambia
                  </option>
                  <option
                    value="ZW"
                  >
                    Zimbabwe
                  </option>
                </select>
                <div
                  aria-hidden="true"
                  class="PhoneInputCountryIcon PhoneInputCountryIcon--border"
                >
                  <img
                    alt="Mexico"
                    class="PhoneInputCountryIconImg"
                    src="https://purecatamphetamine.github.io/country-flag-icons/3x2/MX.svg"
                  />
                </div>
                <div
                  class="PhoneInputCountrySelectArrow"
                />
              </div>
              <input
                autocomplete="tel"
                class="PhoneInputInput"
                placeholder="Ingresa tu número de teléfono"
                type="tel"
                value="+52"
              />
            </div>
          </div>
        </div>
        <button
          class="chakra-button css-bqcyop"
          type="submit"
        >
          Agendar
        </button>
      </fieldset>
    </div>
  </form>
</DocumentFragment>
`;
